File: Main IOS app/TUT Labs/Models/LabSlot.swift
//
//  LabSlot.swift
//  TUT Labs
//
//  Created by Tadiwanashe Songore  on 8/2/25.
//

import Foundation

struct LabStatus: Identifiable {
    var id = UUID()
    var labName: String
    var isOpen: Bool
    var note: String
    var updatedBy: String
    var updatedById: String
    var timestamp: Date
}

struct LabShift: Identifiable {
    var id = UUID()
    var tutorId: String
    var day: String
    var time: String
    var labName: String
}


struct LabSession: Identifiable {
    let id: String
    let labName: String
    let openedById: String
    let openedByName: String
    let openedAt: Date
    let closedById: String?
    let closedByName: String?
    let closedAt: Date?
    let note: String
}

File: Main Android APp/TUTICTLabs/app/src/main/java/com/tadiwanashe/tutictlabs/Models/LabSlot.kt
package com.tadiwanashe.tutictlabs.Models

import java.util.*

// Equivalent to Swift's `var` for mutable properties
data class LabStatus(
    var id: String = UUID.randomUUID().toString(),
    var labName: String = "",
    var isOpen: Boolean = false,
    var note: String = "",
    var updatedBy: String = "",
    var updatedById: String = "",
    var timestamp: Date = Date()
)

data class LabShift(
    var id: String = UUID.randomUUID().toString(),
    var tutorId: String = "",
    var day: String = "",
    var time: String = "",
    var labName: String = ""
)

data class LabSession(
    val id: String,
    val labName: String,
    val openedById: String,
    val openedByName: String,
    val openedAt: Date,
    val closedById: String?,
    val closedByName: String?,
    val closedAt: Date?,
    val note: String
)